activesupport/test/message_encryptor_test.rb

```diff
@@ -132,3 +140,175 @@ def munge(base64_string)
+
+    travel 1.year
+    assert_equal string_message, @encryptor.decrypt_and_verify(payment_related_message, purpose: :payment) 
+
+    travel 1.year + 1.day
+    assert_nil @encryptor.decrypt_and_verify(payment_related_message, purpose: "payment")
+  end
+
+  def test_skip_expires_at_and_expires_in_to_disable_expiration_check
+    payment_related_message = @encryptor.encrypt_and_sign(string_message, purpose: "payment")
+
+    travel 100.years
+    assert_equal string_message, @encryptor.decrypt_and_verify(payment_related_message, purpose: "payment")
+  end
+=begin
+  def test_hash_messages
+    json_encryptor = ActiveSupport::MessageEncryptor.new(@secret, serializer: MessageEncryptorTest::JSONSerializer.new)
+
+    no_purpose = json_encryptor.encrypt_and_sign(hash_message)
+    assert_equal hash_message.stringify_keys, json_encryptor.decrypt_and_verify(no_purpose)
+
+    payment_related_message = json_encryptor.encrypt_and_sign(hash_message, purpose: "payment", expires_at: 1.year.from_now)
+    assert_equal hash_message.stringify_keys, json_encryptor.decrypt_and_verify(payment_related_message, purpose: "payment")
+
+    marshal_encryptor = ActiveSupport::MessageEncryptor.new(@secret, serializer: Marshal)
+
+    no_purpose = marshal_encryptor.encrypt_and_sign(hash_message)
+    assert_equal hash_message, marshal_encryptor.decrypt_and_verify(no_purpose)
+
+    payment_related_message = marshal_encryptor.encrypt_and_sign(hash_message, purpose: "payment", expires_at: 1.year.from_now)
+    assert_equal hash_message, marshal_encryptor.decrypt_and_verify(payment_related_message, purpose: :payment)
+    assert_nil marshal_encryptor.decrypt_and_verify(payment_related_message, purpose: :login)
+  end
+
+  def test_array_messages
+    json_encryptor = ActiveSupport::MessageEncryptor.new(@secret, serializer: MessageEncryptorTest::JSONSerializer.new)
+    
+    no_purpose = json_encryptor.encrypt_and_sign(array_message)
+    assert_equal array_message, json_encryptor.decrypt_and_verify(no_purpose)
+
+    payment_related_message = json_encryptor.encrypt_and_sign(array_message, purpose: "payment", expires_at: 1.year.from_now)
+    assert_equal array_message, json_encryptor.decrypt_and_verify(payment_related_message, purpose: "payment")
+
+    marshal_encryptor = ActiveSupport::MessageEncryptor.new(@secret, serializer: Marshal)
+
+    no_purpose = marshal_encryptor.encrypt_and_sign(array_message)
+    assert_equal array_message, marshal_encryptor.decrypt_and_verify(no_purpose)
+
+    payment_related_message = marshal_encryptor.encrypt_and_sign(array_message, purpose: "payment", expires_at: 1.year.from_now)
+    assert_equal array_message, marshal_encryptor.decrypt_and_verify(payment_related_message, purpose: :payment)
+    assert_nil marshal_encryptor.decrypt_and_verify(payment_related_message, purpose: :login)
+
+    travel 2.years
+    assert_nil marshal_encryptor.decrypt_and_verify(payment_related_message, purpose: "payment")    
+  end
+end
+=end
+
```

Comment:
If the keys deserialized aren't left as strings wouldn't backward compatibility be affected?